src/main/java/com/example/pokebattlez/battle/model/BattlePokemon.java

```diff
@@ -26,15 +28,33 @@
 
     private int currentHp;
 
+    public BattlePokemon(Pokemon pokemon) {
+        id = pokemon.getId();
+        name = pokemon.getName();
+        position = pokemon.getPosition();
+        gender = pokemon.getGender();
+        heldItem = pokemon.getHeldItem();
+        ability = pokemon.getAbility();
+
+        setLevel(pokemon.getLevel());
+        setNature(Nature.valueOf(pokemon.getNature().toUpperCase()));
+
+        moveNames.put(0, pokemon.getMove1());
+        moveNames.put(1, pokemon.getMove2());
+        moveNames.put(2, pokemon.getMove3());
+        moveNames.put(3, pokemon.getMove4());
```

Comment:
You can see that there is a lot of copying, which is usually an indication that something's not necessarily doubled.